xarray/core/dataset.py

```diff
@@ -306,7 +306,7 @@ class Dataset(Mapping, ImplementsDatasetReduce, BaseDataObject,
     groupby_cls = groupby.DatasetGroupBy
 
     def __init__(self, data_vars=None, coords=None, attrs=None,
-                 compat='broadcast_equals'):
+                 compat='broadcast_equals', encoding=None):
```

Comment:
Needs to be documented in the docstring.

Or we could even remove it entirely from the constructor and require setting the property.